query-dsl/src/main/java/org/infinispan/query/dsl/impl/ValueRange.java

```diff
@@ -19,6 +19,12 @@
    private boolean includeUpper = true;
 
    public ValueRange(Object from, Object to) {
+      if (!(from instanceof Comparable)) {
+         throw new IllegalArgumentException("'from' must be an instance of java.lang.Comparable");
```

Comment:
Yes, ideally, but there is no Log in this module yet. I'll add one and see what breaks.